bls/src/main/java/tech/pegasys/teku/bls/mikuli/KeyPair.java

```diff
@@ -36,24 +33,30 @@ public static KeyPair random(final SecureRandom srng) {
     byte[] b = new byte[128];
     srng.nextBytes(b);
     rng.seed(128, b);
-    Scalar secret = new Scalar(BIG.randomnum(curveOrder, rng));
+    // secret key must be between 1 and (curveOrder - 1) inclusive
+    BIG bigSecret = BIG.randomnum(curveOrder.minus(new BIG(1)), rng);
```

Comment:
I couldn't find the documentation but I'm assuming `BIG.randomnum` is exclusive?

Reply:
Afaics, the `BIG.randomnum()` method includes 0, but excludes the upper limit (`curveOrder` here.) in its range of results. My workaround is to subtract one from `curveOrder` and add one to the result. This is hard to test: we might be waiting around to the end of the universe before it generates a number at the end of the range.

Reply:
Yeah, that made sense to me, assuming that the upper bound is exclusive. LOL about waiting around to the end of the universe :D